src/clj/puppetlabs/services/legacy_routes/legacy_routes_core.clj

```diff
@@ -46,21 +85,18 @@
   [handler mount-point api-version request]
   (let [{{environment :environment} :params
          uri :uri
-         query-string               :query-string} request
-        path-info (str "/" (-> (string/split uri #"/" 3) (nth 2)))
-        query-params (if (str/blank? query-string)
-                       {}
-                       (ring-codec/form-decode query-string))]
+         query-string :query-string} request
+        path-info (str "/" (-> (string/split uri #"/" 3) (nth 2)))]
```

Comment:
Actually now that I look back at it, the path is provided by ring. I'm just extracting the first element in the path using a regex. Not sure if `java.net.URI` would help much in that case.